#! /bin/sh /usr/share/dpatch/dpatch-run
## 27doc_sty_noenvpassing.dpatch by <hesso@pool.math.tu-berlin.de>
##
## DP: Document the fact that when calling 'env=var screen app arg' from inside
## DP: screen the environment variable will go up in smoke.

@DPATCH@
diff -Naur screen-4.0.3.orig/doc/screen.1 screen-4.0.3/doc/screen.1
--- screen-4.0.3.orig/doc/screen.1	2008-06-01 23:44:58.000000000 +0200
+++ screen-4.0.3/doc/screen.1	2008-06-01 23:45:00.000000000 +0200
@@ -97,7 +97,9 @@
 manager (specified in the $STY environment variable) who will use it to
 create the new window.
 The above example would start the emacs editor (editing prog.c) and switch
-to its window.
+to its window. - Note that you cannot transport environment variables from
+the invoking shell to the application (emacs in this case), because it is
+forked from the parent screen process, not from the invoking shell.
 .PP
 If \*Q/var/run/utmp\*U is writable by
 .IR screen ,
diff -Naur screen-4.0.3.orig/doc/screen.texinfo screen-4.0.3/doc/screen.texinfo
--- screen-4.0.3.orig/doc/screen.texinfo	2008-06-01 23:44:58.000000000 +0200
+++ screen-4.0.3/doc/screen.texinfo	2008-06-01 23:45:00.000000000 +0200
@@ -175,6 +175,9 @@
 name and its arguments to the window manager (specified in the $STY environment
 variable) who will use it to create the new window.  The above example would 
 start the @code{emacs} editor (editing @file{prog.c}) and switch to its window.
+- Note that you cannot transport environment variables from
+the invoking shell to the application (emacs in this case), because it is
+forked from the parent screen process, not from the invoking shell.
 
 If @file{/var/run/utmp} is writable by @code{screen}, an appropriate record
 will be written to this file for each window, and removed when the
